Obsidian/WorldData/Generators/Overworld/ChunkBuilder.cs

```diff
@@ -144,9 +275,38 @@ internal static void CavesAndOres(GenHelper helper, Chunk chunk)
                     {
                         var oreNoise1 = helper.Noise.Ore(i).GetValue(worldX, y, worldZ);
                         var oreNoise2 = helper.Noise.Ore(i + ores.Length).GetValue(worldX, y, worldZ);
-                        if (oreNoise1 > 1.0 - OreSize && oreNoise2 > 1.0 - OreSize)
+                        if (oreNoise1 > 1.0 - OreSize && oreNoise2 > 1.0 - OreSize && GenerateOreCheck(y, i))
                         {
-                            chunk.SetBlock(worldX, y, worldZ, ores[i]);
+                            if (y <= 0 && y > -64)
+                            {
+                                if (i == 6)
+                                {
+                                    var b = chunk.GetBiome(x, y, z);
+                                    if (new List<Biome> {Biome.WindsweptHills, Biome.WindsweptGravellyHills, Biome.Meadow, Biome.Grove, Biome.SnowySlopes, Biome.FrozenPeaks, Biome.JaggedPeaks, Biome.StonyPeaks}.Contains(b))
+                                    {
+                                        chunk.SetBlock(worldX, y, worldZ, deepores[i]);
+                                    }
+                                }
+                                else
+                                {
+                                    chunk.SetBlock(worldX, y, worldZ, deepores[i]);
+                                }
+                            }
+                            else if (y > 0)
+                            {
+                                if (i == 6)
+                                {
+                                    var b = chunk.GetBiome(x, y, z);
+                                    if (new List<Biome> {Biome.WindsweptHills, Biome.WindsweptGravellyHills, Biome.Meadow, Biome.Grove, Biome.SnowySlopes, Biome.FrozenPeaks, Biome.JaggedPeaks, Biome.StonyPeaks}.Contains(b))
```

Comment:
Instead of creating a list over and over turn this into a variable or private readonly field and call from that instead ^ ^